### ISSUE TITLE
Support Mopeka standard sensor

### ISSUE THREAD

**Issue by o-s-e** [spbrogan/sensor.mopeka_pro_check#2]
Hi,
I was wondering if i would be possible to support the standard sensor from Mopeka?
https://mopeka.com/product/mopeka-check-standard-lpg-sensor-spacers/

I installed the sensor over hacs in my HA installation running on docker.
The container runs in net=host mode and has access to the HCI device, but the sensor fails with:
```
2021-04-28 13:49:56 DEBUG (HCISocketPoller) [bleson] Socket data: len=46, data=04 3e 2b 02 01 03 00 d8 27 2e ff 35 00 1f 1a ff 0d 00 00 02 98 22 e7 8e 21 60 69 47 1a 69 96 49 26 9d 62 84 19 0d 2e 27 d8 03 02 a0 ad b4 
2021-04-28 13:49:56 DEBUG (HCISocketPoller) [bleson] HCIPacket(event_name='LE_Meta_Event', event_code=62, subevent_code=2, data=b'\x01\x03\x00\xd8\'.\xff5\x00\x1f\x1a\xff\r\x00\x00\x02\x98"\xe7\x8e!`iG\x1ai\x96I&\x9db\x84\x19\r.\'\xd8\x03\x02\xa0\xad\xb4', length=42)
2021-04-28 13:49:56 DEBUG (HCISocketPoller) [bleson] BDAddress('00:35:FF:2E:27:D8')
2021-04-28 13:49:56 DEBUG (HCISocketPoller) [bleson] BDAddress('00:35:FF:2E:27:D8')
2021-04-28 13:49:56 ERROR (HCISocketPoller) [mopeka_pro_check.service] Failed to process advertisement from defined sensor.  Exception: Unsupported Data Length (0x1A)
```

**Comment by spbrogan** [comment]
I was told by Mopeka that the standard sensor doesn't have a consistent or documented data format.   That can't be completely true as their app is able to decode them but i took that as they didn't want to describe it to me or maybe can't due to licensing.  I haven't tried to reverse engineer the data format but it looks like you have the stream.  You probably would want to make modifications in the other project first.  Here is the file that you probably would need to extend or make a subclass https://github.com/spbrogan/mopeka_pro_check/blob/main/mopeka_pro_check/advertisement.py  

This project is really about supporting the HA sensor requirements whereas https://github.com/spbrogan/mopeka_pro_check does the BLE advertisement to python object .

**Comment by o-s-e** [comment]
Hi spbrogan, I started playing around with the other project.
I only have the standard sensors, the others are not available here in Europe.
The advertising stream indeed look completely different. 
Unfortunately, I'm not yet familiar enough with BLE to understand the format, yet...
I tried to use hcitools and bettercap to get the GATT's from the sensor, but it doesn't connect... 

I guess it transmits all information in the initial discovery and that's it.

**Comment by spbrogan** [comment]
Yes.  from what i know there is no connection needed.   All info is transmitted in the mfg data of the advertisement.

**Comment by mmiller7** [comment]
I'd also be interested in a way to use with the standard sensor...got one a while back before I knew the pro sensor exists and I'd hate to have to replace it.

If there's something I can do to try and help make this possible I'd be happy to do a test or get some sample data...but I don't know how to do that at the moment.  Also have to figure out some way of getting a bluetooth receiver close enough to the tank to read, phone seems to only work when I'm standing next to it outside.

**Comment by roblamoreaux** [comment]
I am trying to parse the standard sensor data sent I got a couple of them for cheap, and would love to get them into HA. I was dissappointed to find this did not support them, but in trying to find the data format I found this repo (https://github.com/Albertworker1004/Mopeka_React-Native) that seems to have the source for the app in the file at https://github.com/Albertworker1004/Mopeka_React-Native/blob/main/Tank%20Check/src/lib/sensors/sample.ts

I am trying to understand the typescript code for decoding the cc2540 version and getting confused. It seems to have some byteswapping in the parsing of samples that leaves me confused. It looks like the samples are 10 bits with 5 being the delta time and 5 being the amplitude, but the way they seem to be packed into bytes looks like they overlap, when I would expect them to be sequential.

I'm a hardware designer not a software engineer so I am not good with anything but C, and I'd appreciate it if someone could look at the code. I'd like to implement decoding in the ESP32 I have decoding my BLE barbecue thermometer, and if it gets added to this code great, but if not I think I can implement it in ESPHOME Lambdas.
